### PR TITLE
Issue 369,381,380: Validation for Create/Edit FAQ, Create/Edit Offer [fixed]

### LINKED ISSUE #369: Create/Edit Offer Validation Issue
**Create Offer**
- [x] Able to create offer only with [title] and [Active date]
- [x] Error message(Validation) does not displayed on empty field

image

**Edit offer**
- [x] Able to save edit with any empty field (no need Title, content, price, attachment)
- [x] Error message(Validation) does not displayed on empty field

image

### LINKED ISSUE #380: Unable to Edit FAQ contents
- When I try to edit FAQ, it edits title.
image

### LINKED ISSUE #381: Create/Edit FAQ Validation
- [x] Create FAQ
     - no error message displays with null value
     
- [x] Edit FAQ
     - no error message displays with null value

### PR DESCRIPTION
**Pull request by rko7**
**Create Offer Validation**
- validate null value
- validate end date is before start date
image
image
image
image
image

**Edit Offer Validation:**
image

**Create FAQ Validation:**
image

**Edit FAQ Validation:**
image

### CODE REVIEW

**Review by eunbeek — DISMISSED** [review]

**Comment by yuanLeeMidori** [comment]
Is this still in progress? Or ready to review?

**Comment by rko7** [comment]
still in progress.

File: src/VIP/Admin/CreateOffer.js [hunk 1]
```diff
@@ -275,14 +292,23 @@ class CreateOffer extends React.Component {
...
+                  <Col sm={3}>                  
+                    <Form.Control 
+                      controlId="endDate" 
+                      type="date" 
+                      placeholder="end date" 
+                      onChange={this.onEndDateChange.bind(this)} 
+                      isInvalid={this.state.eDateNull || this.state.dateStatus}
+                    />
+                    <Form.Control.Feedback type='invalid'> 
+                      End date is required. Start-date should be before end-date
```
Comment (eunbeek) [inline src/VIP/Admin/CreateOffer.js]: I suggest this sentence might be "Start-date should be before end-date" enough.

Comment (eunbeek) [reply src/VIP/Admin/CreateOffer.js]: Not mandantory suggestion :)

File: src/Request/Admin/EditFAQ.js [hunk 2]
```diff
@@ -61,45 +79,51 @@ class EditFAQ extends React.Component {
...
           title: event.target.value,
         },
-      }));
+        titleNull: true,
+      }))
     }
   }
 
   onContentsChange(event) {
+    this.setState(() => ({
```
Comment (eunbeek) [inline src/Request/Admin/EditFAQ.js:110]: Could you check it one more?
image

Also It is not mandantory

File: src/VIP/Admin/CreateOffer.js (hunk 1 above)
Comment (rko7) [reply src/VIP/Admin/CreateOffer.js]: Oh I will fix like that now

File: src/Request/Admin/EditFAQ.js (hunk 2 above)
Comment (eunbeek) [reply src/Request/Admin/EditFAQ.js:110]: I produced it works well

**Review by eunbeek — DISMISSED** [review]

**Review by yuanLeeMidori — APPROVED** [review]
Great work, thanks!

**Review by eunbeek — APPROVED** [review]

**Pull request merged**